metadata-service/factories/src/main/java/com/linkedin/gms/factory/kafka/DataHubKafkaProducerFactory.java

```diff
@@ -65,7 +65,14 @@
     Map<String, Object> props = properties.buildProducerProperties();
 
     props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, schemaRegistryConfig.getSerializer().getName());
-    props.putAll(schemaRegistryConfig.getProperties());
+
+    // Override KafkaProperties with SchemaRegistryConfig only for non-empty values
+    for (Map.Entry<String, Object> e : schemaRegistryConfig.getProperties().entrySet()) {
```

Comment:
A maybe more concise approach:

schemaRegistryConfig.getProperties().entrySet()
    .stream()
    .filter(entry -> entry.getValue() != null && !entry.toString().isEmpty())
    .forEach(entry -> props.put(entry.getKey(), entry.getValue();

Reply:
If you like it, I like it :)